learn/statistics/survival-case-study/index.qmd

```diff
@@ -0,0 +1,315 @@
+```{r}
+spec_oblique <- rand_forest(mtry = tune(), min_n = tune()) %>% 
+  set_engine("aorsf") %>% 
+  set_mode("censored regression")
+
+wflow_oblique <- workflow() %>% 
+  add_recipe(rec_unknown) %>% 
+  add_model(spec_oblique)
+
+spec_coxnet <- proportional_hazards(penalty = tune()) %>% 
+  set_engine("glmnet") %>% 
+  set_mode("censored regression")
+
+wflow_coxnet <- workflow() %>% 
+  add_recipe(rec_dummies) %>% 
+  add_model(spec_coxnet)
+```
+
+We can tune workflows with any of the `tune_*()` functions such as `tune_grid()` for grid search or `tune_bayes()` for Bayesian optimization. Here we are using grid search for simplicity.
+
+```{r}
+set.seed(1)
+res_oblique <- tune_grid(
+  wflow_oblique,
+  complaints_rset,
+  grid = 10,
+  metrics = metric_set(brier_survival_integrated, brier_survival, roc_auc_survival, concordance_survival),
+  eval_time = seq(0, 300, 30), 
+  control = control_grid(save_workflow = TRUE)
+)
+
+set.seed(1)
+res_coxnet <- tune_grid(
+  wflow_coxnet,
+  complaints_rset,
+  grid = 10,
+  metrics = metric_set(brier_survival_integrated, brier_survival, roc_auc_survival, concordance_survival),
+  eval_time = seq(0, 300, 30), 
+  control = control_grid(save_workflow = TRUE)
+)
+```
+
+So do any of these models perform better than the parametric survival model?
+
+```{r}
+show_best(res_oblique, metric = "brier_survival_integrated", n = 5)
+
+show_best(res_coxnet, metric = "brier_survival_integrated", n = 5)
+```
+
+```{r}
+#| echo: false
+bi_best_cox <- show_best(res_coxnet, metric = "brier_survival_integrated", n = 1) %>% pull(mean)
+bi_best_rf <- show_best(res_oblique, metric = "brier_survival_integrated", n = 1) %>% pull(mean)
+bi_best_survreg <- show_best(res_survreg, metric = "brier_survival_integrated", n = 1) %>% pull(mean)
+
+if (bi_best_cox <= bi_best_rf || bi_best_survreg <= bi_best_rf) {
+  stop("The best model changed.")
+}
+```
+
+The best regularized Cox model performs a little better than the parametric survival model, with an integrated Brier score of `r round(bi_best_cox, 4)` compared to `r round(bi_best_survreg, 4)` for the parametric model. The random forest performs yet a little better with an integrated Brier score of `r round(bi_best_rf, 4)`.
+
+## The final model
+
+We chose the random forest model as the final model. So let's finalize the workflow by replacing the `tune()` placeholders with the best hyperparameters.
+
+```{r}
+param_best <- select_best(res_oblique, metric = "brier_survival_integrated")
+
+wflow_oblique_best <- finalize_workflow(
+  wflow_oblique,
+  param_best
+)
+```
+
+We can now fit the final model on the training data and evaluate it on the test data.
+
+```{r}
+set.seed(2)
+fit_oblique <- last_fit(
+  wflow_oblique_best, 
+  complaints_split,
+  metrics = metric_set(brier_survival_integrated, brier_survival, roc_auc_survival, concordance_survival),
+  eval_time = seq(0, 300, 30)
+)
+
+collect_metrics(fit_oblique) %>% 
+  filter(.metric == "brier_survival_integrated")
+```
+
```

Comment:
It's a bit of code but we can also show how similar the validation and test set results are: 

```r
collect_metrics(res_survreg) %>% 
  filter(.metric == "brier_survival") %>% 
  mutate(Data = "Validation") %>% 
  bind_rows(
    collect_metrics(fit_oblique) %>% 
      filter(.metric == "brier_survival") %>% 
      mutate(Data = "Testing") %>% 
      rename(mean = .estimate)
  ) %>% 
  ggplot(aes(.eval_time, mean, col = Data)) + 
  geom_line() + 
  labs(x = "Evaluation Time", y = "Brier Score")
```

![image](https://github.com/tidymodels/tidymodels.org/assets/5731043/f1560f55-29f3-4eb4-91fc-05f8ffb4a93d)

Reply:
I've changed this to use the Brier scores on the validation set by the oblique forest (rather than the parametric model).